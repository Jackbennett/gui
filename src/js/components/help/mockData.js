export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
