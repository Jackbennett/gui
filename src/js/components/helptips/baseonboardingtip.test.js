import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import BaseOnboardingTip from './baseonboardingtip';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('BaseOnboardingTip Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BaseOnboardingTip anchor={{ left: 1, top: 1, right: 1, bottom: 1 }} component={<div />} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
