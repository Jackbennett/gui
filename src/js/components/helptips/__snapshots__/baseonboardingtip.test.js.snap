// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseOnboardingTip Component renders correctly 1`] = `
<div
  class="onboard-tip"
  style="left: -14px; top: 1px; overflow: initial;"
>
  <a
    class="tooltip onboard-icon bottom"
    currentitem="false"
    data-event="click focus"
    data-event-off="dblclick"
    data-for="onboard-tip-test"
    data-tip="true"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
  </a>
  <div
    class="__react_component_tooltip show place-bottom type-light allow_click content bottom"
    data-id="tooltip"
    id="onboard-tip-test"
    style="left: 0px; top: 0px;"
  >
    <div />
    <div
      class="flexbox"
    >
      <div
        style="flex-grow: 1;"
      />
      <a>
        Dismiss
      </a>
    </div>
  </div>
</div>
`;
